src/parsers/manifestjson.js

```diff
@@ -730,6 +730,26 @@ export default class ManifestJSONParser extends JSONParser {
 
     this.validateRestrictedPermissions();
     this.validateExtensionID();
+    this.ensureHiddenAddonsWithoutBrowserActions();
+  }
+
+  ensureHiddenAddonsWithoutBrowserActions() {
+    // Only privileged add-ons can use the `hidden` manifest property.
+    if (!this.isPrivilegedAddon) {
+      return;
+    }
+
+    if (
+      this.parsedJSON.hidden &&
+      ('action' in this.parsedJSON ||
+        'browser_action' in this.parsedJSON ||
+        // Note: When this was introduced, it was stricter than the Firefox
+        // side because Firefox didn't restrict `page_action` in Bug 1781998.
```

Comment:
Let's file a followup on bugzilla for aligning these restrictions on the Firefox side.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1784254